contracts/LinniaPermissions.sol

```diff
@@ -28,8 +30,8 @@ contract LinniaPermissions is Ownable {
         _;
     }
 
-    modifier onlyFileOwnerOf(bytes32 fileHash) {
-        require(hub.recordsContract().fileOwnerOf(fileHash) == msg.sender);
+    modifier onlyRecordOwnerOf(bytes32 rootHash) {
+        require(hub.recordsContract().recordOwnerOf(rootHash) == msg.sender);
```

Comment:
consider adding the following:
require(rootHash != 0);
Or check if it is a vaild hash.

Reply:
thats unnecessary